gcloud-java-compute/src/main/java/com/google/gcloud/compute/NetworkInterface.java

```diff
@@ -0,0 +1,505 @@
+
+      /**
+       * Sets the type of the access configuration. The only supported value is
+       * {@link Type#ONE_TO_ONE_NAT}.
+       */
+      public Builder type(Type type) {
+        this.type = type;
+        return this;
+      }
+
+      /**
+       * Creates an {@code AccessConfig} object.
+       */
+      public AccessConfig build() {
+        return new AccessConfig(this);
+      }
+    }
+
+    AccessConfig(Builder builder) {
+      this.name = builder.name;
+      this.natIp = builder.natIp;
+      this.type = builder.type;
+    }
+
+    /**
+     * Returns the name of the access configuration.
+     */
+    public String name() {
+      return name;
+    }
+
+    /**
+     * Returns an external IP address associated with this instance.
+     */
+    public String natIp() {
+      return natIp;
+    }
+
+    /**
+     * Returns the type of network access configuration. The only supported value is
+     * {@link Type#ONE_TO_ONE_NAT}.
+     */
+    public Type type() {
+      return type;
+    }
+
+    /**
+     * Returns a builder for the current access configuration.
+     */
+    public Builder toBuilder() {
+      return new Builder(this);
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("name", name)
+          .add("natIp", natIp)
+          .add("type", type)
+          .toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(name, natIp, type);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj == this
+          || obj instanceof AccessConfig
+          && Objects.equals(toPb(), ((AccessConfig) obj).toPb());
+    }
+
+    com.google.api.services.compute.model.AccessConfig toPb() {
+      com.google.api.services.compute.model.AccessConfig accessConfigPb =
+          new com.google.api.services.compute.model.AccessConfig();
+      accessConfigPb.setName(name);
+      accessConfigPb.setNatIP(natIp);
+      if (type != null) {
+        accessConfigPb.setType(type.name());
+      }
+      return accessConfigPb;
+    }
+
+    /**
+     * Returns a builder for an {@code AccessConfig} object.
+     */
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    /**
+     * Returns an {@code AccessConfig} object given the static external IP address.
+     *
+     * @see <a href="https://cloud.google.com/compute/docs/instances-and-network#reservedaddress">
+     *     Static external IP addresses</a>
+     */
+    public static AccessConfig of(String natIp) {
+      return builder().natIp(natIp).build();
+    }
+
+    /**
+     * Returns an {@code AccessConfig} object. IP address for this access configuration will be
+     * taken from a pool of ephemeral addresses.
+     *
+     * @see <a href="https://cloud.google.com/compute/docs/instances-and-network#ephemeraladdress">
+     *     Ephemeral external IP addresses</a>
+     */
+    public static AccessConfig of() {
```

Comment:
Yes it is. If you pass that empty access config an ephemeral address is associated to the instance. Instead, if you pass no access configs at all no address is associated to the instance.